spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/MigrateDeltaLakeTableSparkAction.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.actions;
+
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.delta.BaseMigrateDeltaLakeTableAction;
+import org.apache.iceberg.spark.Spark3Util;
+import org.apache.spark.sql.SparkSession;
+
+/**
+ * Takes a Delta Lake table and attempts to transform it into an Iceberg table in the same location
+ * with the same identifier. Once complete the identifier which previously referred to a non-Iceberg
+ * table will refer to the newly migrated Iceberg table.
+ */
+public class MigrateDeltaLakeTableSparkAction extends BaseMigrateDeltaLakeTableAction {
+  MigrateDeltaLakeTableSparkAction(
+      SparkSession spark,
+      String deltaTableLocation,
```

Comment:
In the Spark implementation, we can derive the delta table location from a delta lake table name, because you can load the table by name and get it's location. That is probably more user friendly to use compared to a table location.

Reply:
Looks like the benefit of using the location is that it does not require installing the Delta plugin. I agree that's a good benefit, looking at the demo it is not hard to get the location in Spark either.